### PR TITLE
chore(api): hide generateClient from api/server subpath

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

* Fixed missing `api` in `adapter-nextjs` packages.json `files`
* Hide `generateClient` from `aws-amplify/api/server`

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/adapter-nextjs/src/api/generateServerClient.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { generateClient as internalGenerateClient } from '@aws-amplify/api/server';
+import { generateClient as internalGenerateClient } from '@aws-amplify/api/internals';
```
Comment (iartemiev) [inline packages/adapter-nextjs/src/api/generateServerClient.ts:4]: There are 2 versions of `generateClient`. We want the server one here

Comment (HuiSF) [reply packages/adapter-nextjs/src/api/generateServerClient.ts:4]: I moved the server edition of `generateClient` into `packages/api/src/internals/InternalAPI.ts` which is being exported from '@aws-amplify/api/internals' in this PR.

Should rename the function to make it clearer it's the server edition?

Comment (iartemiev) [reply packages/adapter-nextjs/src/api/generateServerClient.ts:4]: Nevermind. That's not in `main` yet. But only in `api-v6-models`. We'll need to merge this carefully not to overwrite it

**Review by jimblanc — DISMISSED** [review]

**Review by AllanZhengYP — APPROVED** [review]
Make sense for the `/api` path. Defer to Ivan's comments on the `/server` to `/internals` change

**Review by jimblanc — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Pull request merged**